src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1387,12 +1387,12 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
 
     ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
+    uct_dc_mlx5_iface_dci_put(iface, dci_index);
```

Comment:
maybe add comments why need put before failed and process_pending after reset? Otherwise it may be changed again in the future